widgets/weather_all.lua

```diff
@@ -17,6 +18,15 @@ local helpers = require"vicious.helpers"
 -- vicious.widgets.weather
 local weather_all = {}
 
+-- copied from http://lua-users.org/wiki/TimeZone
+local function get_timezone_offset()
+    local ts = os.time()
+    local utcdate   = os.date("!*t", ts)
+    local localdate = os.date("*t", ts)
+    localdate.isdst = false -- this is the trick
```

Comment:
I've just (mentally) evaluate this and realize that this is not supposed to be used here. The original function was to compute timezone, whilst what we want is actual offset. To sum up all requests, I drafted a patch as follow:

```diff
diff --git a/widgets/weather_all.lua b/widgets/weather_all.lua
index 6bd4cfc..ba7e886 100644
--- a/widgets/weather_all.lua
+++ b/widgets/weather_all.lua
@@ -6,6 +6,7 @@
 -- {{{ Grab environment
 local tonumber = tonumber
 local math = { ceil = math.ceil }
+local os = { date = os.date, difftime = os.difftime, time = os.time }
 local string = { match = string.match }
 
 local spawn = require"vicious.spawn"
@@ -17,12 +18,15 @@ local helpers = require"vicious.helpers"
 -- vicious.widgets.weather
 local weather_all = {}
 
+-- Actual time offset with day-saving taken onto account
+local TIME_OFFSET = os.difftime(os.time, os.time(os.date"!*t"))
 
 -- {{{ Weather widget type
 local function parse(stdout, stderr, exitreason, exitcode)
     -- Initialize function tables
     local _weather = {
         ["{city}"]    = "N/A",
+        ["{when}"]    = "N/A",
         ["{wind}"]    = "N/A",
         ["{windmph}"] = "N/A",
         ["{windkmh}"] = "N/A",
@@ -58,6 +62,18 @@ local function parse(stdout, stderr, exitreason, exitcode)
     _weather["{press}"]   = -- Pressure in hPa
        string.match(stdout, "Pressure[%s].+%((.+)[%s]hPa%)") or _weather["{press}"]
 
+    local year, month, day, hour, min =
+        string.match(stdout, "(%d%d%d%d).(%d%d).(%d%d) (%d%d)(%d%d) UTC")
+    if year ~= nil then
+        _weather["{when}"] = TIME_OFFSET + os.time{
+            year = year,
+            month = month,
+            day = day,
+            hour = hour,
+            minute = minute
+        }
+    end
+
     -- Wind speed in km/h if MPH was available
     if _weather["{windmph}"] ~= "N/A" then
        _weather["{windmph}"] = tonumber(_weather["{windmph}"])
```

Day saving is not available in my tropical country, so feel free to discuss if you think I'm wrong.

Reply:
Right; this will mean the offset will be wrong until users restart awesome at DST.  If you are ok with that it's fine with me.

Reply:
I totally missed that d-: My point was that `isdst` should not be changed, and my attempt to save a few instructions failed miserably. I guess `time_offset` still needs to be a function, i.e.

```lua
local function time_offset()
    return os.difftime(os.time(), os.time(os.date"!*t"))
end
```

Reply:
yep that seems good to me.  You can apply this change or I can modify my patch; whichever you prefer.

Reply:
> I can modify my patch

Please do, then we can squash your commits together and it'd make a more meaningful `blame`.